public/app/features/plugins/sandbox/distortion_map.ts

```diff
@@ -0,0 +1,73 @@
+type DistortionMap = Map<unknown, unknown>;
+const generalDistortionMap: DistortionMap = new Map();
+
+function failToSet() {
+  throw new Error('Plugins are not allowed to set sandboxed properties');
+}
+
+// sets distortion to protect iframe elements
+function distordIframeAttributes(distortions: DistortionMap) {
+  const iframeHtmlForbiddenProperties = ['contentDocument', 'contentWindow', 'src', 'srcdoc', 'srcObject', 'srcset'];
+
+  for (const property of iframeHtmlForbiddenProperties) {
+    const descriptor = Object.getOwnPropertyDescriptor(HTMLIFrameElement.prototype, property);
+    if (descriptor) {
+      function fail() {
+        throw new Error('iframe.' + property + ' is not allowed in sandboxed plugins');
+      }
+      if (descriptor.value) {
+        distortions.set(descriptor.value, fail);
+      }
+      if (descriptor.set) {
+        distortions.set(descriptor.set, fail);
+      }
+      if (descriptor.get) {
+        distortions.set(descriptor.get, fail);
+      }
+    }
+  }
+}
+
+function distordConsole(distortions: DistortionMap) {
```

Comment:
Typo? (I see it in some other function names as well.)
```suggestion
function distortConsole(distortions: DistortionMap) {
```